library/src/main/java/com/afollestad/materialcamera/internal/Camera2Fragment.java

```diff
@@ -275,23 +275,48 @@ public static Camera2Fragment newInstance() {
     return fragment;
   }
 
-  private static Size chooseVideoSize(BaseCaptureInterface ci, Size[] choices) {
-    Size backupSize = null;
-    for (Size size : choices) {
-      if (size.getHeight() <= ci.videoPreferredHeight()) {
-        if (size.getWidth() == size.getHeight() * ci.videoPreferredAspect()) {
-          return size;
-        }
-        if (ci.videoPreferredHeight() >= size.getHeight()) {
-          backupSize = size;
-        }
+  public static Size getSizeWithClosestRatio(Size[] sizes, int targetWidth, int targetHeight) {
+    if (sizes == null) return null;
+
+    double MIN_TOLERANCE = 100;
```

Comment:
constant

Reply:
unit?

Reply:
the unit is the same as the rest of the variables, so I don't think is necessary